pkg/registry/core/shoot/strategy.go

```diff
@@ -172,6 +168,11 @@ func (shootStatusStrategy) PrepareForUpdate(ctx context.Context, obj, old runtim
 	newShoot := obj.(*core.Shoot)
 	oldShoot := old.(*core.Shoot)
 	newShoot.Spec = oldShoot.Spec
+
+	if lastOperation := newShoot.Status.LastOperation; lastOperation != nil && lastOperation.Type == core.LastOperationTypeRestore &&
```

Comment:
Why was that moved out of the `mustIncreaseGeneration` function?

Reply:
In the previous function shoot.oldStatus is set to the shoot.newStatus meaning lastOperation was old not Restore=Pending, so it needed one more update in order to increase the generation.

Reply:
I see, thank you!